New parameter type for coordinate operations

nyalldawson opened the pull request:
Allows selection of the proj coordinate operation to use when reprojecting between two CRSes

Currently not used in any algorithms, but the plan is to add it to the vector and raster reprojection algorithms (but only when running on proj 6 builds)

nyalldawson committed:
Move guts of datum transformation dialog to a new standalone widget
QgsCoordinateOperationWidget

This widget shows a list of the available operations between a source
and destination CRS along with their properties

nyalldawson committed:
Avoid duplicate signals, add unit test

nyalldawson committed:
Use QgsCoordinateOperationWidget in QgsDatumTransformDialog

nyalldawson committed:
Fix changing source or dest CRS when editing a coordinate operation
fails to correctly save the new operation

nyalldawson committed:
Allow double-clicking on coordinate operation to quickly select them

nyalldawson committed:
[API][processing] New parameter type for coordinate operations

Allows selection of the proj coordinate operation to use when
reprojecting between two CRSes

nyalldawson committed:
Move variant to CRS resolution out to QgsProcessingUtils for wider
usability

nyalldawson committed:
Allow algorithms to link to multiple depandant parameters in modeler scene

nyalldawson committed:
Fix warnings

nyalldawson committed:
[processing] Add gui widget wrappers for coordinate operation parameter

nyalldawson committed:
Unit tests

nyalldawson committed:
Fix build on crufty proj

alexbruy reviewed (approved):
Looks good to me

nirvn commented:
@nyalldawson , could this also be useful to add in the save [vector/raster] layer as, since we have a CRS selector there?

nyalldawson committed:
Dumbass Travis build

nyalldawson committed:
Add test for proj < 6 logic

nyalldawson committed:
No margins

nyalldawson committed:
Correctly resolve project layers to CRS values in widget

nyalldawson committed:
Only expose coordinate operation parameter type on Proj 6+ builds

nyalldawson committed:
Add API to auto set QgsCoordinateOperationWidget selection to match
a QgsCoordinateTransformContext's state

nyalldawson committed:
Auto select operation in processing widget based on project transform context

nyalldawson committed:
[FEATURE][processing] Allow selection of specific coordinate operation
to use when reprojecting vector layers via the "Reproject Layer" algorithm

On proj 6+ builds only!

Allows users to set a specific operation to use for a particular reprojection
task, instead of always forcing use of the current project's transformation
settings.

Useful when reprojecting a particular layer and control over the exact transformation
pipeline is required.

nyalldawson committed:
Skip a test on travis

nyalldawson committed:
Disable test on proj < 6

nyalldawson committed:
Don't run a test under proj 6
